### PR TITLE
Document security of serve_dir router

### PR DESCRIPTION
**Pull request by halvko**
As someone new to the library, I worry about the security builtin routes, and here I had to check the (well documented) source to find what i was looking for. This PR adds the information to the route documentation itself.

### CODE REVIEW

**Review by jbr — APPROVED** [review]
Seems reasonable to me. Do we 406 in response to paths outside of the served first whether or not there's a file at the path? If so, we might want to clarify that. (And if not, that's a bug)

File: src/route.rs [hunk 1]
```diff
@@ -118,6 +118,12 @@ impl<'a, State: Clone + Send + Sync + 'static> Route<'a, State> {
     /// Each file will be streamed from disk, and a mime type will be determined
     /// based on magic bytes.
     ///
+    /// # Security
+    /// 
+    /// This handler ensures no folders outside the requested folder can be served,
+    /// and attempts to access such files will return StatusCode::Forbidden to the
+    /// caller
```
Comment (Fishrock123) [inline src/route.rs:125]:
```suggestion
    /// caller.
```

File: src/route.rs [hunk 2]
```diff
@@ -118,6 +118,12 @@ impl<'a, State: Clone + Send + Sync + 'static> Route<'a, State> {
     /// Each file will be streamed from disk, and a mime type will be determined
     /// based on magic bytes.
     ///
+    /// # Security
+    /// 
+    /// This handler ensures no folders outside the requested folder can be served,
+    /// and attempts to access such files will return StatusCode::Forbidden to the
```
Comment (Fishrock123) [inline src/route.rs:124]:
```suggestion
    /// and attempts to access such files will return `StatusCode::Forbidden` to the
```

**Comment by halvko** [comment]
> Seems reasonable to me. Do we 406 in response to paths outside of the served first whether or not there's a file at the path? If so, we might want to clarify that. (And if not, that's a bug)

Yes we do, I'll add it to the PR

File: src/route.rs (hunk 1 above)
Comment (halvko) [reply src/route.rs]: Oops, thanks ^.^

File: src/route.rs (hunk 2 above)
Comment (halvko) [reply src/route.rs]: Thanks

**Comment by halvko** [comment]
@jbr I was a little quick there, we 403: Forbidden - which i believe is correct behavior. From http-types:
403 Forbidden: "The client does not have acces rights to the content [...]"
406 NotAcceptable: "[... the server] doesn't find any content that conforms to the criteria given by the user agent"

**Comment by halvko** [comment]
> @halvko Could you rebase to pick up clippy lint fixes from `main`?

Is now done

**Comment by Fishrock123** [comment]
@halvko Sorry, one more thing - could you please run `cargo fmt --all`?

**Comment by halvko** [comment]
Should now be fixed

**Comment by Fishrock123** [comment]
Thanks!

**Pull request merged**